jscomp/super_errors/super_typecore.ml

```diff
@@ -167,7 +167,21 @@ let report_error env ppf = function
       (_, {desc = Tarrow _}) ::
       (_, {desc = Tconstr (Pdot (Pdot(Pident {name = "Js"},"Fn",_),_,_),_,_)}) :: _
     ) ->
-    fprintf ppf "This function is a curried function where an uncurried function is expected"
+    fprintf ppf "This function is curried; we expected an uncurried function, like f(. a, b)"
+  | Expr_type_clash (
+      (_, {desc = Tconstr (Pdot (Pdot(Pident {name = "Js"},"Fn",_),a,_),_,_)}) ::
+      (_, {desc = Tconstr (Pdot (Pdot(Pident {name = "Js"},"Fn",_),b,_),_,_)}) :: _
+    ) when a <> b ->
+    let extractArity s =
```

Comment:
Is this too weird? Should I just pattern match all the way from `arity0` to `arity22`?

Reply:
`pattern match all the way from arity0 to arity22` that would be an explosion of pattern match

Reply:
why it is still sound ? since we own the  Js.Fn module which only contains  type definitions for `arity*`

Reply:
- can you extract this into a toplevel function parameterized by `a` and `b` so that it can be reused later (for other syntaxes as well)?
- this looks like the same pattern match in loc 185-188, so you should merge it there
- I also think that `a <> b` seems to be not necessary since when `a = b`, there should be no such error message